Browserify compatibility, dependency improvements, delimiters and tagOnBlur props

bruderstein opened the pull request:
This is a combined pull request for several very small changes. 

1. Adds compatibility with browserify
2. Adds the default css under a fixed "css" directory (rather than a client having to include the file from examples)
3. Moves react to be a [peerDependency](http://blog.nodejs.org/2013/02/07/peer-dependencies/) (>=0.12.1 <0.13)
4. Adds key property to TagComponents
5. delimiters as prop, so they are customisable
6. tagOnBlur prop, so if the user enters text, then tags away, a tag is automatically created (when prop is true)

thmain commented:
@bruderstein Everything seems cool. Except for one tiny thing. The ```grunt dev``` wont work anymore since the example files rely on react. We would at least need to add it as a dev dependency for the demos.

bruderstein commented:
Ok, done.  I didn't realise peerDeps wouldn't be installed when they're in the root. Seems like a reasonable thing for npm to do.

I've rebased on your master again, and added react in as a devDependency in the same commit as it becomes a peerDep.

thmain commented:
:+1:
